script.js

```diff
@@ -97,45 +125,82 @@ function setPlatform(platform) {
     currentPlatform = platform;
 }
 
-function generateRegionInfo(name, ip, fqdn, port) {
+function setPortIfDefault(protocol) {
+    const oldPort = protocol == "http" ? "443" : "22000";
+    const newPort = protocol == "http" ? "22000" : "443";
+    const portField = document.getElementById("port");
+    if (portField.value == oldPort) {
+        portField.value = newPort;
+    }
+}
+
+function generateRegionInfo(name, fqdn, port, protocol) {
     const regions = [
         // Add default regions so they also show up in the menu
         {
-            $type: "DnsRegionInfo, Assembly-CSharp",
-            Fqdn: "na.mm.among.us",
-            DefaultIp: "50.116.1.42",
-            Port: 22023,
+            $type: "StaticHttpRegionInfo, Assembly-CSharp",
             Name: "North America",
+            PingServer: "matchmaker.among.us",
+            Servers: [
+                {
+                    Name: "Http-1",
+                    Ip: "https://matchmaker.among.us",
+                    Port: 443,
+                    UseDtls: true,
+                    Players: 0,
+                    ConnectionFailures: 0,
+                },
+            ],
             TranslateName: 289,
-            UseDtls: true,
         },
         {
-            $type: "DnsRegionInfo, Assembly-CSharp",
-            Fqdn: "eu.mm.among.us",
-            DefaultIp: "172.105.251.170",
-            Port: 22023,
+            $type: "StaticHttpRegionInfo, Assembly-CSharp",
             Name: "Europe",
+            PingServer: "matchmaker-eu.among.us",
+            Servers: [
+                {
+                    Name: "Http-1",
+                    Ip: "https://matchmaker-eu.among.us",
+                    Port: 443,
+                    UseDtls: true,
+                    Players: 0,
+                    ConnectionFailures: 0,
+                },
+            ],
             TranslateName: 290,
-            UseDtls: true,
         },
         {
-            $type: "DnsRegionInfo, Assembly-CSharp",
-            Fqdn: "as.mm.among.us",
-            DefaultIp: "139.162.111.196",
-            Port: 22023,
+            $type: "StaticHttpRegionInfo, Assembly-CSharp",
             Name: "Asia",
+            PingServer: "matchmaker-as.among.us",
+            Servers: [
+                {
+                    Name: "Http-1",
+                    Ip: "https://matchmaker-as.among.us",
+                    Port: 443,
+                    UseDtls: true,
+                    Players: 0,
+                    ConnectionFailures: 0,
+                },
+            ],
             TranslateName: 291,
-            UseDtls: true,
         },
```

Comment:
I don't think we need these anymore, seems like they get added automatically when missing

Reply:
They do get automatically added, but the lack of official servers can confuse some people who are not aware of that happening.

I shared a template json file in the discord a few days ago and I already received a few comments from people that were missing the original servers. Adding the bytes doesn't hurt and as it reduces confusion I think we should leave them in

Reply:
I mean more the maintaining burden if among us ever changes the addresses or anything.
As for confusion we could add a comment in json if you really want? Personally I don't think it's really a problem.

Reply:
I don't expect regular changes to the addresses, and even if upstream changes these addresses, old region files will be silently updated anyway. New addresses will likely also require changes in our region file generator